core/src/main/java/org/hisp/dhis/android/network/organisationunit/OrganisationUnitDTO.kt

```diff
@@ -74,12 +77,37 @@ internal data class OrganisationUnitDTO(
                 closedDate?.let { closedDate(it) }
             }
             .level(level)
-            .geometry(geometry?.toDomain())
+            .geometry(evaluateGeometry())
             .programs(programs?.map { ObjectWithUid.create(it.uid) })
             .programs(programs?.map { ObjectWithUid.create(it.uid) })
             .dataSets(dataSets?.map { ObjectWithUid.create(it.uid) })
             .organisationUnitGroups(organisationUnitGroups?.map { it.toDomain() })
-            .displayNamePath(displayNamePath)
+            .displayNamePath(evaluateDisplayNamePath())
             .build()
     }
+
+    private fun evaluateDisplayNamePath(): List<String?>? {
```

Comment:
The displayNamePath is evaluated here, so a lot classes can be removed from the domain.